src/vswhere.lib/Formatter.cpp

```diff
@@ -556,6 +560,10 @@ HRESULT Formatter::GetProductId(_In_ ISetupInstance* pInstance, _Out_ VARIANT* p
                 *pvtProductId = vt.Detach();
             }
         }
+        else if (SUCCEEDED(hr))
```

Comment:
The idea was to actually let the child `hr` propagate back to the caller. This changes the behavior. It is intended to be `__fallthrough`.

Reply:
Couldn't find a __fallthrough macro for this case, just for switch statements, so I suppressed them here